atlp-rwanda/e-commerce-lydia-32-bn
# A buyer can update their cart and also their total items count will b…

Certainly! Here's the PR template using the specified format:

---

### What does this PR do?
This PR implements the functionality to handle updates to the buyer's cart, including storing the updated cart in the buyer's session and calculating the new cart total.

### Description of Task to be completed?
- Handle `post` requests to update the cart via the cart ID.
- Store the updated cart in the buyer's session.
- Retrieve product details (**name**, **price**, **image**) from the database for each item in the updated cart.
- Calculate the new cart total by summing the prices of the items in the updated cart.
- Send an updated cart confirmation message to the frontend to be displayed to the buyer.

### How should this be manually tested?
1. Send a `POST` request to update the cart with a valid cart ID and updated quantities.
2. Verify that the updated cart is stored in the buyer's session.
3. Check that the product details for each item in the updated cart are correctly retrieved from the database.
4. Confirm that the new cart total is accurately calculated.
5. Ensure an updated cart confirmation message is sent to the frontend.

### Any background context you want to provide?
#### N/A

### Screenshots (if appropriate)
#### N/A

### Questions:
#### N/A

### What are the relevant pivotal tracker stories?
#### N/A

---

This template should help in creating a comprehensive and clear pull request for updating the cart functionality.
